core/api/test/integration/services/lnd-service.spec.ts

```diff
@@ -209,160 +214,281 @@ afterAll(async () => {
+
+      const paid = await lndService.payInvoiceViaPaymentDetails({
+        decodedInvoice: lnInvoice,
+        btcPaymentAmount,
+        maxFeeAmount: undefined,
+      })
+      expect(paid).toBeInstanceOf(PaymentRejectedByDestinationError)
     })
-    if (paid instanceof Error) throw paid
-    expect(paid.revealedPreImage).toHaveLength(64)
 
-    const retryPaid = await lndService.payInvoiceViaPaymentDetails({
-      decodedInvoice: lnInvoice,
-      btcPaymentAmount,
-      maxFeeAmount: undefined,
+    it("pay invoice with High CLTV Delta", async () => {
+      // Create invoice
+      const { request } = await createInvoice({
+        lnd: lndOutside1,
+        tokens: amountInvoice,
+        cltv_delta: 200,
+      })
+      const lnInvoice = decodeInvoice(request)
+      if (lnInvoice instanceof Error) throw lnInvoice
+
+      const paid = await lndService.payInvoiceViaPaymentDetails({
+        decodedInvoice: lnInvoice,
+        btcPaymentAmount,
+        maxFeeAmount: undefined,
+      })
+      if (paid instanceof Error) throw paid
+      expect(paid.revealedPreImage).toHaveLength(64)
     })
-    expect(retryPaid).toBeInstanceOf(LnAlreadyPaidError)
-  })
 
-  it("fails to pay when channel capacity exceeded", async () => {
-    const { request } = await createInvoice({ lnd: lndOutside1, tokens: 1500000 })
-    const lnInvoice = decodeInvoice(request)
-    if (lnInvoice instanceof Error) throw lnInvoice
+    it("pays high fee route with no max limit", async () => {
+      const lnInvoice = await createPrivateInvoice({
+        lnd: lndOutside2,
+      })
+
+      const paid = await lndService.payInvoiceViaPaymentDetails({
+        decodedInvoice: lnInvoice,
+        btcPaymentAmount,
+        maxFeeAmount: undefined,
+      })
+      if (paid instanceof Error) throw paid
+      expect(paid.revealedPreImage).toHaveLength(64)
+      expect(paid.roundedUpFee).toEqual(
+        Number(btcPaymentAmount.amount) * ROUTE_PPM_PERCENT,
+      )
+    })
 
-    const paid = await lndService.payInvoiceViaPaymentDetails({
-      decodedInvoice: lnInvoice,
-      btcPaymentAmount,
-      maxFeeAmount: undefined,
+    it("fails to pay high fee route with max limit set", async () => {
+      const lnInvoice = await createPrivateInvoice({
+        lnd: lndOutside2,
+      })
+
+      const paid = await lndService.payInvoiceViaPaymentDetails({
+        decodedInvoice: lnInvoice,
+        btcPaymentAmount,
+        maxFeeAmount: LnFees().maxProtocolAndBankFee(btcPaymentAmount),
+      })
+      expect(paid).toBeInstanceOf(RouteNotFoundError)
     })
-    expect(paid).toBeInstanceOf(PaymentRejectedByDestinationError)
-  })
 
-  it("pay invoice with High CLTV Delta", async () => {
-    // Create invoice
-    const { request } = await createInvoice({
-      lnd: lndOutside1,
-      tokens: amountInvoice,
-      cltv_delta: 200,
+    it("probes across route with fee higher than payment amount", async () => {
```

Comment:
probably is better `it("fails` but not a blocker

Reply:
https://github.com/GaloyMoney/galoy/pull/3694/commits/dbe2248c4049bc54d967e6c1735eb815ba905dcb